test/integration-no-mock/config-management/env-config-create.js

```diff
@@ -0,0 +1,404 @@
+      }
+    ], done);
+  });
+
+  it('internal collections plus system collections, internal hooks and roles should succeed', (done) => {
+    const collList = EnvHelper.buildValidInternalCollectionsList(2, true);
+    const rolesList = ConfigManagementHelper.roles.buildValidRolesList(2);
+
+    // set complex permissions for a collection
+    const roleName = rolesList[0].name;
+    const collWithComplexPermissions = collList[0];
+    collWithComplexPermissions.permissions = {
+      'all-users': {
+        create: 'always',
+        read: 'grant',
+        update: 'always',
+        delete: 'grant'
+      },
+      [roleName]: {
+        create: 'grant',
+        update: 'entity'
+      }
+    };
+
+    const collHooks = {
+      [collList[0].name]: {
+        onPreSave: {
+          type: 'internal'
+        }
+      },
+      [collList[1].name]: {
+        onPostFetch: {
+          type: 'internal',
+          code: 'function onPostFetch(request, response, modules) {\nconsole.log("Here");\n  response.continue();\n}'
+        }
+      }
+    };
+
+    const env = {
+      schemaVersion: '1.0.0',
+      configType: 'environment',
+      collections: ConfigManagementHelper.common.buildConfigEntityFromList(collList),
+      collectionHooks: collHooks,
+      roles: ConfigManagementHelper.common.buildConfigEntityFromList(rolesList)
+    };
+
+    const envName = randomStrings.envName();
+    const rolesNameIdPairs = {};
+    let envId;
+    let actualRoles;
+
+    async.series([
+      (next) => {
+        EnvHelper.createFromConfig(envName, env, appName, (err, id) => {
+          if (err) {
+            return next(err);
+          }
+
+          envId = id;
+          next();
+        });
+      },
+      (next) => {
+        EnvHelper.assertEnvOnly(env, envId, envName, next);
+      },
+      (next) => {
+        ApiService.roles.get(envId, null, (err, data) => {
+          if (err) {
+            return next(err);
+          }
+
+          actualRoles = data;
+
+          const expectedRolesCount = rolesList.length;
+          expect(expectedRolesCount).to.equal(actualRoles.length);
+
+          for (let i = 0; i < expectedRolesCount; i += 1) {
+            const expected = rolesList[i];
+            const actual = actualRoles.find(x => x.name === expected.name);
+            if (!actual) {
+              return next(new Error(`Failed to find role with name '${expected.name}'.`));
+            }
+
+            rolesNameIdPairs[actual.name] = actual._id;
+            expect(actual.description).to.equal(expected.description);
+          }
+
+          next();
+        });
+      },
+      (next) => {
+        EnvHelper.assertCollections(envId, collList, collList.length, rolesNameIdPairs, next);
+      },
+      (next) => {
+        EnvHelper.assertAllCollHooks(envId, collHooks, next);
+      }
+    ], done);
+  });
+
+  it('common code, internal endpoints, groups and push settings should succeed', (done) => {
+    const commonCode = {
```

Comment:
The test does not assert that the common code is created correctly and will not fail if it isn't.

Reply:
Nice catch :clap: